pkg/logs/input/journald/tailer_util.go

```diff
@@ -0,0 +1,63 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package journald
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+)
+
+const (
+	tagSeparator            = ":"
+	imageTagKey             = "short_image" + tagSeparator
+	defaultShortNameValue   = "docker"
+	baseCacheExpiration     = 20 * time.Minute
+	expirationSpreadSeconds = 120
+)
+
+// Get docker image short from dockerId and list of tags
+// In case of miss in cache, we parse from tag list.
+// We could use the util/Docker.Inspect + ImageID resolution or Regexp on config.Image
+// But this way is simple and probably fast enough (~300ns for 100 items) on miss
+func getDockerImageShortName(containerID string, tags []string) (string, bool) {
+	cacheKey := getImageCacheKey(containerID)
+	cacheContent, hit := cache.Cache.Get(cacheKey)
+	if hit {
+		value, castOk := cacheContent.(string)
```

Comment:
Do we want to override the entry if the cast is not ok ? I think it's fine, worth adding a comment stating that the key is specific enough to avoid collision.

Reply:
Added a comment to clearly state we'll override anyway